org_fedora_oscap/rule_handling.py

```diff
@@ -668,3 +682,50 @@ def eval_rules(self, ksdata, storage, report_only=False):
             return []
 
     # nothing to be reverted for now
+
+
+class KdumpRules(RuleHandler):
+    """Simple class holding data from the rules affecting the kdump addon."""
+
+    def __init__(self):
+        """Constructor setting the initial value of attributes."""
+
+        self._kdump_enabled = True
+
+    def kdump_enabled(self):
+        self._kdump_enabled = False
+
+    def __str__(self):
+        """Standard method useful for debugging and testing."""
+
+        ret = "kdump"
+
+        if not self._kdump_enabled:
+            ret += " --disable"
+
+        return ret
+
+    def eval_rules(self, ksdata, storage, report_only=False):
+        """:see: RuleHandler.eval_rules"""
+
+        messages = []
+
+        if not self._kdump_enabled:
+            msg = _("Kdump will be disabled on startup")
+
+            messages.append(RuleMessage(self.__class__,
+                                        common.MESSAGE_TYPE_INFO, msg))
+
+            try:
+                ksdata.addons.com_redhat_kdump.enabled = self._kdump_enabled
+            except AttributeError:
+                log.warning("com_redhat_kdump is not installed. "
+                            "Skipping disabling kdump")
+
+        return messages
+
+    def revert_changes(self, ksdata, storage):
+        """:see: RuleHander.revert_changes"""
+
+        if not self._kdump_enabled:
+            ksdata.addons.com_redhat_kdump.enabled = True
```

Comment:
this looks suspicious to me, shouldn't we rather save the previous state in eval_rules and the apply it here if it's not None?

Reply:
@mpreisler I am pretty sure that we are just reverting everything that we did in eval_rules. I don't think that we are saving states from eval_rules at all. However, `self._kdump_enabled` should be reset back to `True` here.